nmdc_runtime/site/translation/neon_surface_water_translator.py

```diff
@@ -280,51 +287,69 @@ def _translate_library_preparation(
             start_date=_get_value_or_none(library_preparation_row, "seqCollectDate"),
             end_date=_get_value_or_none(library_preparation_row, "seqProcessedDate"),
             processing_institution=processing_institution,
+            type="nmdc:LibraryPreparation",
         )
 
-    def _translate_omics_processing(
+    def _get_instrument_id(self, instrument_model: Union[str | None]) -> str:
+        if not instrument_model:
+            raise ValueError(
+                f"instrument_model '{instrument_model}' could not be found in the NEON-NMDC instrument mapping TSV file."
+            )
+
+        df = self.neon_nmdc_instrument_map_df
+        matching_row = df[
+            df["NEON sequencingMethod"].str.contains(instrument_model, case=False)
+        ]
+
+        if not matching_row.empty:
+            nmdc_instrument_id = matching_row["NMDC instrument_set id"].values[0]
+            return nmdc_instrument_id
```

Comment:
I guess it doesn't really need to be taken care of in this PR, but it seems odd that this method is duplicated in three different classes. Couldn't it be inherited from some (possibly new) base class? 

Same for a lot of code in these three classes, but this new method stands out in particular.

Reply:
That's a really good pattern to follow, but didn't cross my mind. Probably have a `NeonTranslator` base class and then have 3 translator classes inherited from it. I'll make a note of this, and address it in a different PR if/when NEON translators refactor becomes a priority? It's kind of a low priority translator because I don't see it being run again. Just a one time pull.